### PR TITLE
Moved tests to CircleCI

### PR DESCRIPTION
**Pull request by gp510**
- Moved tests to CircleCI
- Removed Codeship files
- Fixed readme typo
- Set default sourcetype

### CODE REVIEW

File: .circleci/unit_tests.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+set -e
+
+mkdir /home/circleci/.go_workspace/bin
+curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
+dep ensure
+
+echo "Running Golang unit tests..."
+
+go get github.com/Sirupsen/logrus
```
Comment (luckyj5) [inline .circleci/unit_tests.sh]: Why do we still need to install deps separately if we are running `dep ensure`?

File: splunk_test.go [hunk 2]
```diff
@@ -1411,7 +1412,7 @@ func TestCannotSendAfterClose(t *testing.T) {
 		t.Fatal("Driver should not allow to send messages after close")
 	}
 
-	if len(hec.messages) != 1 {
+	if !(len(hec.messages) > 0) {
```
Comment (luckyj5) [inline splunk_test.go]: why is the logic modified here from length != 1 condition?

**Review by luckyj5 — DISMISSED** [review]
Functional tests are missing from CI pipelines. 
Also, please squash commits and resubmit.

File: splunk_logger.go [hunk 3]
```diff
@@ -568,6 +568,7 @@ func telemetry(info logger.Info, l *splunkLogger, sourceType string, splunkForma
 		url:       l.hec.url,
 	}
 
+	time.Sleep(5 * time.Second)
```
Comment (dtregonning) [inline splunk_logger.go:571]: What is the sleep for?

Comment (gp510) [reply splunk_logger.go:571]: The sleep prevents the telemetry function from throwing off the event count for the unit tests

File: .circleci/unit_tests.sh (hunk 1 above)
Comment (gp510) [reply .circleci/unit_tests.sh]: Removed

File: splunk_test.go (hunk 2 above)
Comment (gp510) [reply splunk_test.go]: That was a mistake. Reverted to the original test

**Review by dtregonning — APPROVED** [review]
Squash all commits and clean up when PR goes in.
Ensure all tests are passing (Green Check) Circle CI before cutting release

**Pull request merged**